File: android/app/src/main/java/com/vxin/app/data/repository/MomentRepository.kt
package com.vxin.app.data.repository

import com.vxin.app.data.api.MomentApi
import com.vxin.app.data.model.CreateMomentBody
import com.vxin.app.data.model.Moment
import com.vxin.app.data.model.MomentComment
import com.vxin.app.data.model.MomentCommentBody
import okhttp3.MultipartBody
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class MomentRepository @Inject constructor(
    private val momentApi: MomentApi,
    socketManager: com.vxin.app.core.realtime.SocketManager,
) {
    /** 朋友圈实时事件（新动态/点赞/评论） */
    val momentEvents = socketManager.momentEvents

    suspend fun timeline(limit: Int = 20, offset: Int = 0): List<Moment> = momentApi.timeline(limit, offset)

    suspend fun create(
        content: String,
        images: List<String>,
        visibility: String,
        visibleTo: List<String> = emptyList(),
    ): Moment =
        momentApi.create(CreateMomentBody(content, images, visibility, visibleTo))

    suspend fun uploadImages(parts: List<MultipartBody.Part>): List<String> =
        momentApi.uploadImages(parts).urls

    suspend fun like(id: String) = momentApi.like(id)

    suspend fun comment(id: String, content: String, replyToUser: String = ""): MomentComment =
        momentApi.comment(id, MomentCommentBody(content, replyToUser))

    suspend fun delete(id: String) = momentApi.delete(id)

    suspend fun deleteComment(commentId: String) = momentApi.deleteComment(commentId)

    suspend fun comments(id: String, limit: Int = 50, offset: Int = 0) =
        momentApi.comments(id, limit, offset)
}

File: ios/Vxin/Data/Repositories/MomentRepository.swift
import Foundation
import Combine

private struct CreateMomentBody: Encodable { let content: String; let images: [String]; let visibility: String; let visibleTo: [String] }
private struct CommentBody: Encodable { let content: String }

final class MomentRepository {
    static let shared = MomentRepository()
    private init() {}

    private let api = APIClient.shared

    var eventsPublisher: AnyPublisher<Void, Never> { SocketService.shared.moments.eraseToAnyPublisher() }

    func timeline(limit: Int = 20, offset: Int = 0) async throws -> [Moment] {
        try await api.send("api/moments?limit=\(limit)&offset=\(offset)")
    }

    func create(content: String, images: [String], visibility: String, visibleTo: [String] = []) async throws -> Moment {
        try await api.send("api/moments", method: "POST", body: CreateMomentBody(content: content, images: images, visibility: visibility, visibleTo: visibleTo))
    }

    func uploadImages(_ datas: [(data: Data, name: String)]) async throws -> [String] {
        // 逐张上传后合并（后端 /images 支持多图，但 APIClient.upload 为单文件，这里顺序上传）
        var urls: [String] = []
        for d in datas {
            let res: MomentImagesResponse = try await api.upload("api/moments/images", fileData: d.data, fileName: d.name, mimeType: "image/jpeg", fieldName: "images")
            urls.append(contentsOf: res.urls)
        }
        return urls
    }

    func like(_ id: String) async throws -> MomentLikeResponse {
        try await api.send("api/moments/\(id)/like", method: "POST")
    }

    func comment(_ id: String, content: String) async throws -> MomentComment {
        try await api.send("api/moments/\(id)/comment", method: "POST", body: CommentBody(content: content))
    }

    func delete(_ id: String) async throws {
        let _: EmptyResponse = try await api.send("api/moments/\(id)", method: "DELETE")
    }

    func deleteComment(_ commentId: String) async throws {
        let _: EmptyResponse = try await api.send("api/moments/comments/\(commentId)", method: "DELETE")
    }

    func comments(_ id: String, limit: Int = 50, offset: Int = 0) async throws -> CommentPage {
        try await api.send("api/moments/\(id)/comments?limit=\(limit)&offset=\(offset)")
    }
}
